
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtGui import QPixmap


class Ui_mainWindow(object):

    def setupUi(self, mainWindow):
        mainWindow.setObjectName("mainWindow")
        mainWindow.setEnabled(True)
        mainWindow.resize(904, 608)
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(255, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(214, 214, 212))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(214, 214, 212))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(214, 214, 212))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(214, 214, 212))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        mainWindow.setPalette(palette)
        mainWindow.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        mainWindow.setContextMenuPolicy(QtCore.Qt.DefaultContextMenu)
        mainWindow.setTabShape(QtWidgets.QTabWidget.Rounded)
        self.centralwidget = QtWidgets.QWidget(mainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.Node1_label = QtWidgets.QLabel(self.centralwidget)
        self.Node1_label.setGeometry(QtCore.QRect(20, 130, 61, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.Node1_label.setFont(font)
        self.Node1_label.setObjectName("Node1_label")
        self.Node2_label = QtWidgets.QLabel(self.centralwidget)
        self.Node2_label.setGeometry(QtCore.QRect(230, 130, 61, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.Node2_label.setFont(font)
        self.Node2_label.setObjectName("Node2_label")
        self.Weight_label = QtWidgets.QLabel(self.centralwidget)
        self.Weight_label.setGeometry(QtCore.QRect(130, 130, 61, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.Weight_label.setFont(font)
        self.Weight_label.setObjectName("Weight_label")
        self.AddEdge_button = QtWidgets.QPushButton(self.centralwidget)
        self.AddEdge_button.setGeometry(QtCore.QRect(310, 170, 101, 28))
        palette = QtGui.QPalette()
        self.AddEdge_button.setPalette(palette)
        font = QtGui.QFont()
        font.setPointSize(9)
        self.AddEdge_button.setFont(font)
        self.AddEdge_button.setStyleSheet("\n"
"\n"
"\n"
"\n"
"\n"
"")
        self.AddEdge_button.setCheckable(False)
        self.AddEdge_button.setDefault(False)
        self.AddEdge_button.setFlat(False)
        self.AddEdge_button.setObjectName("AddEdge_button")
        self.Start_label = QtWidgets.QLabel(self.centralwidget)
        self.Start_label.setGeometry(QtCore.QRect(30, 420, 51, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.Start_label.setFont(font)
        self.Start_label.setObjectName("Start_label")
        self.End_label = QtWidgets.QLabel(self.centralwidget)
        self.End_label.setGeometry(QtCore.QRect(240, 420, 61, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.End_label.setFont(font)
        self.End_label.setObjectName("End_label")
        self.comboBox = QtWidgets.QComboBox(self.centralwidget)
        self.comboBox.setGeometry(QtCore.QRect(330, 70, 171, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.comboBox.setFont(font)
        self.comboBox.setObjectName("comboBox")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.chooseSA_label = QtWidgets.QLabel(self.centralwidget)
        self.chooseSA_label.setGeometry(QtCore.QRect(330, 40, 181, 20))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.chooseSA_label.setFont(font)
        self.chooseSA_label.setObjectName("chooseSA_label")
        self.Graph_output = QtWidgets.QGraphicsView(self.centralwidget)
        self.Graph_output.setGeometry(QtCore.QRect(440, 180, 441, 281))
        self.Graph_output.setObjectName("Graph_output")

        self.Path_output = QtWidgets.QTextBrowser(self.centralwidget)
        self.Path_output.setGeometry(QtCore.QRect(20, 370, 381, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.Path_output.setFont(font)
        self.Path_output.setObjectName("Path_output")
        self.FinalPath_label = QtWidgets.QLabel(self.centralwidget)
        self.FinalPath_label.setGeometry(QtCore.QRect(190, 320, 81, 41))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.FinalPath_label.setFont(font)
        self.FinalPath_label.setObjectName("FinalPath_label")
        self.Graph_label = QtWidgets.QLabel(self.centralwidget)
        self.Graph_label.setGeometry(QtCore.QRect(630, 140, 61, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.Graph_label.setFont(font)
        self.Graph_label.setObjectName("Graph_label")
        self.Heuristic_label = QtWidgets.QLabel(self.centralwidget)
        self.Heuristic_label.setEnabled(True)
        self.Heuristic_label.setGeometry(QtCore.QRect(180, 220, 61, 20))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.Heuristic_label.setFont(font)
        self.Heuristic_label.setObjectName("Heuristic_label")
        self.Draw_button = QtWidgets.QPushButton(self.centralwidget)
        self.Draw_button.setGeometry(QtCore.QRect(620, 480, 71, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.Draw_button.setFont(font)
        self.Draw_button.setObjectName("Draw_button")
        self.Heuristic_Input = QtWidgets.QTextEdit(self.centralwidget)
        self.Heuristic_Input.setEnabled(True)
        self.Heuristic_Input.setGeometry(QtCore.QRect(180, 250, 41, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.Heuristic_Input.setFont(font)
        self.Heuristic_Input.setObjectName("Heuristic_Input")
        self.Start_input = QtWidgets.QTextEdit(self.centralwidget)
        self.Start_input.setGeometry(QtCore.QRect(30, 460, 41, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.Start_input.setFont(font)
        self.Start_input.setObjectName("Start_input")
        self.End_input = QtWidgets.QTextEdit(self.centralwidget)
        self.End_input.setGeometry(QtCore.QRect(240, 460, 41, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.End_input.setFont(font)
        self.End_input.setObjectName("End_input")
        self.Node2Input = QtWidgets.QTextEdit(self.centralwidget)
        self.Node2Input.setGeometry(QtCore.QRect(240, 170, 41, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.Node2Input.setFont(font)
        self.Node2Input.setObjectName("Node2Input")
        self.Node1Input = QtWidgets.QTextEdit(self.centralwidget)
        self.Node1Input.setGeometry(QtCore.QRect(20, 170, 41, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.Node1Input.setFont(font)
        self.Node1Input.setObjectName("Node1Input")
        self.WeightInput = QtWidgets.QTextEdit(self.centralwidget)
        self.WeightInput.setGeometry(QtCore.QRect(130, 170, 41, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.WeightInput.setFont(font)
        self.WeightInput.setObjectName("WeightInput")
        self.graphIcon_label = QtWidgets.QLabel(self.centralwidget)
        self.graphIcon_label.setGeometry(QtCore.QRect(600, 140, 31, 21))
        self.graphIcon_label.setText("")
        self.graphIcon_label.setPixmap(QtGui.QPixmap("Graph.png"))
        self.graphIcon_label.setScaledContents(True)
        self.graphIcon_label.setObjectName("graphIcon_label")
        self.Path_label = QtWidgets.QLabel(self.centralwidget)
        self.Path_label.setGeometry(QtCore.QRect(150, 330, 31, 21))
        self.Path_label.setText("")
        self.Path_label.setPixmap(QtGui.QPixmap("path.png"))
        self.Path_label.setScaledContents(True)
        self.Path_label.setObjectName("Path_label")
        self.AddHeuristic_button = QtWidgets.QPushButton(self.centralwidget)
        self.AddHeuristic_button.setEnabled(True)
        self.AddHeuristic_button.setGeometry(QtCore.QRect(310, 240, 101, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.AddHeuristic_button.setFont(font)
        self.AddHeuristic_button.setObjectName("AddHeuristic_button")
        self.submitEnd_button = QtWidgets.QPushButton(self.centralwidget)
        self.submitEnd_button.setGeometry(QtCore.QRect(110, 500, 101, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.submitEnd_button.setFont(font)
        self.submitEnd_button.setObjectName("submitEnd_button")
        self.Hnode_Input = QtWidgets.QTextEdit(self.centralwidget)
        self.Hnode_Input.setEnabled(True)
        self.Hnode_Input.setGeometry(QtCore.QRect(80, 250, 41, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.Hnode_Input.setFont(font)
        self.Hnode_Input.setObjectName("Hnode_Input")
        self.Hnode_label = QtWidgets.QLabel(self.centralwidget)
        self.Hnode_label.setEnabled(True)
        self.Hnode_label.setGeometry(QtCore.QRect(80, 220, 51, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.Hnode_label.setFont(font)
        self.Hnode_label.setObjectName("Hnode_label")
        self.line_3 = QtWidgets.QFrame(self.centralwidget)
        self.line_3.setGeometry(QtCore.QRect(60, 180, 181, 16))
        self.line_3.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_3.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_3.setObjectName("line_3")
        self.line = QtWidgets.QFrame(self.centralwidget)
        self.line.setGeometry(QtCore.QRect(0, 300, 441, 20))
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.line_4 = QtWidgets.QFrame(self.centralwidget)
        self.line_4.setGeometry(QtCore.QRect(70, 470, 171, 16))
        self.line_4.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_4.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_4.setObjectName("line_4")
        self.checkBox = QtWidgets.QCheckBox(self.centralwidget)
        self.checkBox.setGeometry(QtCore.QRect(350, 110, 141, 20))
        self.checkBox.setObjectName("checkBox")
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setGeometry(QtCore.QRect(300, 460, 121, 31))
        self.pushButton.setObjectName("pushButton")
        self.radioButton = QtWidgets.QRadioButton(self.centralwidget)
        self.radioButton.setGeometry(QtCore.QRect(310, 200, 101, 20))
        self.radioButton.setObjectName("radioButton")
        self.clearButton = QtWidgets.QPushButton(self.centralwidget)
        self.clearButton.setGeometry(QtCore.QRect(380, 540, 111, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.clearButton.setFont(font)
        self.clearButton.setObjectName("pushButton_2")
        self.Node1_label.raise_()
        self.Node2_label.raise_()
        self.Weight_label.raise_()
        self.AddEdge_button.raise_()
        self.Start_label.raise_()
        self.End_label.raise_()
        self.comboBox.raise_()
        self.chooseSA_label.raise_()
        self.Graph_output.raise_()
        self.Path_output.raise_()
        self.Graph_label.raise_()
        self.Heuristic_label.raise_()
        self.Draw_button.raise_()
        self.Heuristic_Input.raise_()
        self.End_input.raise_()
        self.Node1Input.raise_()
        self.graphIcon_label.raise_()
        self.Path_label.raise_()
        self.AddHeuristic_button.raise_()
        self.submitEnd_button.raise_()
        self.Hnode_Input.raise_()
        self.Hnode_label.raise_()
        self.line_3.raise_()
        self.line.raise_()
        self.Node2Input.raise_()
        self.line_4.raise_()
        self.Start_input.raise_()
        self.WeightInput.raise_()
        self.checkBox.raise_()
        self.pushButton.raise_()
        self.radioButton.raise_()
        self.clearButton.raise_()
        self.FinalPath_label.raise_()
        mainWindow.setCentralWidget(self.centralwidget)

        self.retranslateUi(mainWindow)
        QtCore.QMetaObject.connectSlotsByName(mainWindow)

    def retranslateUi(self, mainWindow):
        _translate = QtCore.QCoreApplication.translate
        mainWindow.setWindowTitle(_translate("mainWindow", "AI searching Algorithms"))
        self.Node1_label.setText(_translate("mainWindow", "Node 1"))
        self.Node2_label.setText(_translate("mainWindow", "Node 2"))
        self.Weight_label.setText(_translate("mainWindow", "Weight"))
        self.AddEdge_button.setText(_translate("mainWindow", "Add Edge"))
        self.Start_label.setText(_translate("mainWindow", "Start"))
        self.End_label.setText(_translate("mainWindow", "End"))
        self.comboBox.setItemText(0, _translate("mainWindow", "None"))
        self.comboBox.setItemText(1, _translate("mainWindow", "Breadth First Search"))
        self.comboBox.setItemText(2, _translate("mainWindow", "Depth First Search"))
        self.comboBox.setItemText(3, _translate("mainWindow", "Uniform Cost Search"))
        self.comboBox.setItemText(4, _translate("mainWindow", "Greedy Search"))
        self.comboBox.setItemText(5, _translate("mainWindow", "A* search"))
        self.chooseSA_label.setText(_translate("mainWindow", "Choose searching Algorithm"))
        self.FinalPath_label.setText(_translate("mainWindow", "Final Path"))
        self.Graph_label.setText(_translate("mainWindow", "Graph"))
        self.Heuristic_label.setText(_translate("mainWindow", "Heuristic"))
        self.Draw_button.setText(_translate("mainWindow", "Draw "))
        self.AddHeuristic_button.setText(_translate("mainWindow", "Add Heuristic"))
        self.submitEnd_button.setText(_translate("mainWindow", "Get Path"))
        self.Hnode_label.setText(_translate("mainWindow", "Node"))
        self.checkBox.setText(_translate("mainWindow", "Directed Nodes"))
        self.pushButton.setText(_translate("mainWindow", "Add New Goal"))
        self.radioButton.setText(_translate("mainWindow", "Tree Finished"))
        self.clearButton.setText(_translate("mainWindow", "Clear"))
        self.Draw_button.pressed.connect( lambda: self.graph("draw"))
        self.clearButton.pressed.connect (lambda: self.graph("c"))
    def graph(self,x):
        scene = QtWidgets.QGraphicsScene(self)
        pixmap = QPixmap("fig.png")
        item = QtWidgets.QGraphicsPixmapItem(pixmap)
        if x == "draw":
            scene.addItem(item)
            self.Graph_output.setScene(scene)
            self.Graph_output.fitInView(item)
            self.radioButton.setEnabled(False)
        else:
            scene.removeItem(item)
            self.Graph_output.setScene(scene)


if __name__ == "__main__":
    import sys

    app = QtWidgets.QApplication(sys.argv)
    mainWindow = QtWidgets.QMainWindow()
    ui = Ui_mainWindow()
    ui.setupUi(mainWindow)
    mainWindow.show()
    sys.exit(app.exec_())
